src/main/java/org/folio/service/transformationfields/TransformationFieldsConfig.java

```diff
@@ -24,21 +24,21 @@
-  ELECTRONIC_ACCESS_MATERIALS_SPECIFIED("electronic.access.materialsSpecification", "$.{recordType}.electronicAccess[?(@.relationshipId=={id})].materialsSpecification", ELECTRONIC_ACCESS_RELATIONSHIPS),
-  ELECTRONIC_ACCESS_PUBLICNOTE("electronic.access.publicNote", "$.{recordType}.electronicAccess[?(@.relationshipId=={id})].publicNote", ELECTRONIC_ACCESS_RELATIONSHIPS),
+  ELECTRONIC_ACCESS_URI("electronic.access.uri", "$.{recordType}.electronicAccess[?(@.relationshipId=='{id}')].uri", ELECTRONIC_ACCESS_RELATIONSHIPS),
+  ELECTRONIC_ACCESS_LINKTEXT("electronic.access.linkText", "$.{recordType}.electronicAccess[?(@.relationshipId=='{id}')].linkText", ELECTRONIC_ACCESS_RELATIONSHIPS),
+  ELECTRONIC_ACCESS_MATERIALS_SPECIFIED("electronic.access.materialsSpecification", "$.{recordType}.electronicAccess[?(@.relationshipId=='{id}')].materialsSpecification", ELECTRONIC_ACCESS_RELATIONSHIPS),
+  ELECTRONIC_ACCESS_PUBLICNOTE("electronic.access.publicNote", "$.{recordType}.electronicAccess[?(@.relationshipId=='{id}')].publicNote", ELECTRONIC_ACCESS_RELATIONSHIPS),
   ELECTRONIC_ACCESS_URI_DEFAULT("electronic.access.uri", "$.{recordType}.electronicAccess[?(!(@.relationshipId) || @.relationshipId == null)].uri"),
   ELECTRONIC_ACCESS_LINKTEXT_DEFAULT("electronic.access.linkText", "$.{recordType}.electronicAccess[?(!(@.relationshipId) || @.relationshipId == null)].linkText"),
   ELECTRONIC_ACCESS_MATERIALS_SPECIFIED_DEFAULT("electronic.access.materialsSpecification", "$.{recordType}.electronicAccess[?(!(@.relationshipId) || @.relationshipId == null)].materialsSpecification"),
   ELECTRONIC_ACCESS_PUBLICNOTE_DEFAULT("electronic.access.publicNote", "$.{recordType}.electronicAccess[?(!(@.relationshipId) || @.relationshipId == null)].publicNote"),
 
   //Instance specific fields
-  ALTERNATIVE_TITLES("alternativeTitleTypeId", "$.instance[*].alternativeTitles[?(@.alternativeTitleTypeId=={id})]", ALTERNATIVE_TITLE_TYPES),
-  CONTRIBUTOR_NAME_TYPE("contributorNameTypeId", "$.instance[*].contributorNameTypeId[?(@.contributorNameTypeId=={id} && ?(!(@.primary) || @.primary == false))]", CONTRIBUTOR_NAME_TYPES),
-  CONTRIBUTOR_NAME_TYPE_PRIMARY("contributorNameTypeId.primary", "$.instance[*].contributorNameTypeId[?(@.contributorNameTypeId=={id} && ?((@.primary) && @.primary == true))]", CONTRIBUTOR_NAME_TYPES),
+  ALTERNATIVE_TITLES("alternativeTitleTypeName", "$.instance.alternativeTitles[?(@.alternativeTitleTypeId=='{id}')].name", ALTERNATIVE_TITLE_TYPES),
+  CONTRIBUTOR_NAME("contributorName", "$.instance.contributors[?(@.contributorNameTypeId=='{id}' && (!(@.primary) || @.primary == false))].name", CONTRIBUTOR_NAME_TYPES),
+  CONTRIBUTOR_NAME_PRIMARY("contributorName.primary", "$.instance.contributors[?(@.contributorNameTypeId=='{id}' && ((@.primary) && @.primary == true))].name", CONTRIBUTOR_NAME_TYPES),
```

Comment:
I found that true/false statements in jsonpath should be in single quotes '' to evaluate path correctly

Reply:
And we also found that quotes for {id} isn't required

Reply:
Quotes are needed Illia
<img width="1680" alt="Screen Shot 2020-08-20 at 8 44 36 AM" src="https://user-images.githubusercontent.com/41596468/90771550-73f19880-e2c1-11ea-9b4d-95857ac0a708.png">
<img width="1676" alt="Screen Shot 2020-08-20 at 8 44 45 AM" src="https://user-images.githubusercontent.com/41596468/90771553-75bb5c00-e2c1-11ea-8593-cc1c1a442622.png">

Reply:
Check here https://github.com/json-path/JsonPath#filter-operators
> String literals must be enclosed by single or double quotes ([?(@.color == 'blue')] or [?(@.color == "blue")]).

UUID is a String in our schema

Reply:
Yeah, you are right, sorry for wrong comment